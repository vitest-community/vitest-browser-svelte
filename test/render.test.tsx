import { expect, test } from 'vitest'
import { page } from 'vitest/browser'
import { render } from 'vitest-browser-svelte'
import HelloWorld from './fixtures/HelloWorld.svelte'
import Counter from './fixtures/Counter.svelte'

test('renders simple component', async () => {
  const screen = render(HelloWorld)
  await expect.element(page.getByText('Hello World')).toBeVisible()
  expect(screen.container).toMatchSnapshot()
})

test('renders counter', async () => {
  const screen = render(Counter, {
    initialCount: 1,
  })

  await expect.element(screen.getByText('Count is 1')).toBeVisible()
  await screen.getByRole('button', { name: 'Increment' }).click()
  await expect.element(screen.getByText('Count is 2')).toBeVisible()
})
