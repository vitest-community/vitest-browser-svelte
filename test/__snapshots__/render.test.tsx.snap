// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders simple component 1`] = `
<div>
  <div>
    Hello World
  </div>
  
</div>
`;
